### ISSUE TITLE
Native PostgreSQL Timestamp Format Incorrectly Escaped

### ISSUE THREAD

**Issue by ericdrobinson** [porsager/postgres#222]
I've run into issues with both sending and receiving "native" PostgreSQL Timestamp strings. While I'm able to sidestep the _receiving_ case by following the suggestion [here](https://github.com/porsager/postgres/issues/161#issuecomment-801031062) of casting the timestamp to a string type via `::text`, I've not found a suitable solution for the _sending_ case.

Specifically, a timestamp generated by Postgres via a query of the following format:

```sql
SELECT transaction_timestamp()::text as timestamp
```

will look like this:

```
2021-09-03 05:48:18.382017+00
```

Unfortunately, when we pass that string to PostgeSQL using `postgres`, we receive the following error:

```
PostgresError: invalid input syntax for type timestamp: ""2021-09-03 05:48:18"."382017+00""
```

A little time with a debugger showed that the offending function is [this one](https://github.com/porsager/postgres/blob/a02de6756910e7baf73e151270fb1404079ad4a7/lib/types.js#L68):

```js
module.exports.escape = function escape(str) {
  return '"' + str.replace(/"/g, '""').replace(/\./g, '"."') + '"'
}
```

Out of curiosity, why is this function globally replacing the `.` character with `"."` in the first place?

Regardless of how this is resolved, I think adding some documentation about how values passed in with `sql(...)` are escaped. While the documentation currently says:

> To enable dynamic queries in a safe way, the `sql` function doubles as a regular function which escapes any value properly.

it doesn't provide any explanation as to what sorts of escaping it's handling/performing. Also, as this says "any value properly", is it safe to assume that I've stumbled across a bug?

**Comment by porsager** [comment]
Can you add the js code query you're seeing this with?

**Comment by ericdrobinson** [comment]
This is a simple repro case:

```js
await sql`
WITH ts AS (SELECT transaction_timestamp() AS timestamp)
SELECT timestamp
FROM ts
WHERE timestamp > '${sql('2021-09-03 06:00:42.705651+00')}'`;
```

That works perfectly in pgAdmin. Breaks in `postgres`.

**Comment by ericdrobinson** [comment]
I've also confirmed that for this use case, at least, removing the `escape` logic from `postgres` allows this query to work successfully.

**Comment by porsager** [comment]
Right, the documentation must be confusing at some point. You only use `sql(string)` inside tagged template literals to reference columns.

Your query above should simply be 
```js
await sql`
WITH ts AS (SELECT transaction_timestamp() AS timestamp)
SELECT timestamp
FROM ts
WHERE timestamp > ${ '2021-09-03 06:00:42.705651+00' }`;
```

**Comment by ericdrobinson** [comment]
That won't work.

**Comment by porsager** [comment]
Why not?

**Comment by ericdrobinson** [comment]
We need some form of string interpolation because we pass that value in by variable. So at best we have this:

```js
const time = '2021-09-03 06:00:42.705651+00';
await sql`
WITH ts AS (SELECT transaction_timestamp() AS timestamp)
SELECT timestamp
FROM ts
WHERE timestamp > ${time}`;
```

If we do _that_, however, we get the following error:

```
PostgresError: invalid input syntax for type timestamp with time zone: "$1"
```

**Comment by ericdrobinson** [comment]
I.e. we can't hard-code a timestamp into our source code, which is what you wrote above :(

**Comment by porsager** [comment]
Well what you just wrote works fine too.

**Comment by porsager** [comment]
Let's take this to https://gitter.im/porsager/postgres perhaps 🙂

**Comment by ericdrobinson** [comment]
@porsager Thanks very much for helping work through the issue over on Gitter.

For those following along, I had added an extra pair of `'` quotation marks around the interpolated timestamp position. Behind the scenes, Postgres sees `$1` and is passed a parameter array to fill things in. It "magically" infers the type while still in that array, so there's no need for the quotes. In fact, they cause the call to _break_ because suddenly Postgres sees `'$1'`, which is a string as opposed to a parameter. 🤦

Glad we got it working, though!
